### ISSUE TITLE
Don't load issues if title load fails

### ISSUE THREAD

**Issue by jechols** [open-oni/open-oni#414]
Maybe we need to simply handle errors better in the title loader, I'm not sure.  But it seems that when trying to load a title, failures get logged but just sort of pass through, allowing issues to load even though a title may be missing critical data.

It's possible this is only a bug on the Django 2.2 code which I'm currently implementing, but I believe I recall similar problems when I had invalid marc data in our Django 1.11 instance.

**Comment by jechols** [comment]
I think we should actually consider a transaction in the title loader.  If anything fails, we don't want the title saved.  Right now there are *two* calls to `title.save()` as well as calls to other `.save()` methods.  We should persist all this title data only if all of it works.

Fixing this might actually fix the problem I had in #411 - I just verified that an incomplete title (specifically one with no languages) will yield a Solr 400 error.

**Comment by jechols** [comment]
Verified that a transaction wrapping the main title loading function fixes this behavior.  It causes a crash that isn't very obvious, so it may not be the ideal fix, but it at least prevents getting bad data in the database.

Diff:

```patch
diff --git a/core/title_loader.py b/core/title_loader.py
index a19ed94..9f4b501 100644
--- a/core/title_loader.py
+++ b/core/title_loader.py
@@ -6,7 +6,7 @@
 from time import time, strptime

 from pymarc import map_xml, record_to_xml
-from django.db import reset_queries
+from django.db import reset_queries, transaction
 from django.utils import timezone

 from core import models
@@ -58,6 +58,7 @@ def load_record(record):

         map_xml(load_record, urllib.request.urlopen(location))

+    @transaction.atomic
     def load_bib(self, record):
         title = None
```

**Comment by jechols** [comment]
This was accidentally fixed in the same branch as the Django 2.2 upgrade
